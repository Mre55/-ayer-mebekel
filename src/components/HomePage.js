import React, { useEffect, useState } from "react";

import { fetchWeatherWithoutDispatch } from "../redux/weather/weatherReducer";
import { useDispatch } from "react-redux";

import DetailsPage from "./DetailsPage";
import Form from "./Form";
import CurrentPollutionData from "./CurrentPollutionData";
import store from "../redux/configureStore";
import { fetchWeatherDispatch } from "../redux/weather/weatherReducer";
import backIcon from "../images/backIcon.png";

const HomePage = () => {
  const [newPageData, setNewPageData] = useState();
  const [firstPageData, setFirstPageData] = useState();
  const [secondPageData, setSecondPageData] = useState();
  const [thirdPageData, setThirdPageData] = useState();
  const [fourthPageData, setFourthPageData] = useState();
  const [selectedCity, setSelectedCity] = useState();
  const [detailPage, setDetailPage] = useState(false);

  const dispatch = useDispatch();
  const fetchNewData = async (lat, long) => {
    const withoutDi = await fetchWeatherWithoutDispatch(lat, long);
    dispatch(fetchWeatherDispatch());
    const [first, second, third, fourth] = await Promise.all(
      [
        [50, 50],
        [8, 37],
        [40, 40],
        [30, 20],
      ].map((coord) => fetchWeatherWithoutDispatch(coord[0], coord[1]))
    );
    setNewPageData(withoutDi);
    setFirstPageData(first);
    setSecondPageData(second);
    setThirdPageData(third);
    setFourthPageData(fourth);
  };

  const seeMore = (city) => {
    setSelectedCity(city);
    setDetailPage(!detailPage);
  };

  const getSelectedCityData = () => {
    switch (selectedCity) {
      case "Hawassa":
        return secondPageData;
      case "Addis Ababa":
        return firstPageData;
      case "Hossana":
        return thirdPageData;
      case "London":
        return fourthPageData;
      case "New":
        return newPageData;
      default:
        return firstPageData;
    }
  };

  const selectedCityData = getSelectedCityData();

  useEffect(() => {
    if (Object.keys(store.getState().weatherReducer).length === 0) {
      dispatch(fetchWeatherDispatch());
    }
  }, []);

  return (
    <div>
      <div className={`${detailPage ? "hidden" : ""}`}>
        <Form seeMore={seeMore} fetchNewData={fetchNewData} />

        <div className={`${detailPage ? "hidden" : ""}`}>
          <CurrentPollutionData />
        </div>
      </div>
      <div className={`${detailPage ? "" : "hidden"}`}>
        <div className="flex items-center gap-14 bg-[#35538c] text-white px-3 ">
          <button className="w-10" onClick={() => seeMore()} type="button">
            <img src={backIcon} alt="back icon" />
          </button>
          <p className="text-sm text-white py-2 w-screen">
            DATA ABOUT POLLUTING GASES
          </p>
        </div>
        {selectedCityData?.coord && (
          <DetailsPage
            coord={selectedCityData.coord}
            list={selectedCityData.list}
          />
        )}
      </div>
    </div>
  );
};

export default HomePage;
