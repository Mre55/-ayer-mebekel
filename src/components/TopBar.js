import React from 'react';

const TopBar = () => (
  <nav className="">
    <div className="flex items-end justify-center py-2 ring-4 bg-[#4369b2]">
      <h2 className="text-2xl font-bold text-white">የአየር ብክለት </h2>
      <h2 className="text-lg font-bold text-white"> (Air pollution)</h2>
    </div>
  </nav>
);
export default TopBar;
