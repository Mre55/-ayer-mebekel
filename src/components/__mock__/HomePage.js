import React from 'react';

import { useDispatch } from 'react-redux';
import { fetchWeatherWithInput } from '../../redux/weather/weatherReducer';

import Form from '../Form';
import backIcon from '../../images/backIcon.png';

const HomePage = () => {
  const dispatch = useDispatch();

  const fetchDataByInput = async (lat, long) => {
    dispatch(fetchWeatherWithInput(lat, long));
  };

  const seeMore = (city) => {
    if (city === 'Addis Ababa') {
      dispatch(fetchWeatherWithInput(4, 5));
    } else if (city === 'Hawassa') {
      dispatch(fetchWeatherWithInput(8, 37));
    } else if (city === 'Hossana') {
      dispatch(fetchWeatherWithInput(22, 44));
    } else if (city === 'London') {
      dispatch(fetchWeatherWithInput(17, 9));
    }
  };

  return (
    <div>
      <div className={'\'detailPage\' ? \'hidden\' : \'\''}>
        <Form seeMore={seeMore} fetchDataByInput={fetchDataByInput} />

        <div className={'\'detailPage\' ? \'hidden\' : \'\''} />
      </div>
      <div className={'\'detailPage\' ? \'\' : \'hidden\''}>
        <div className="flex items-center gap-14 bg-[#35538c] text-white px-3 ">
          <button className="w-10" onClick={() => seeMore()} type="button">
            <img src={backIcon} alt="back icon" />
          </button>
          <p className="text-sm text-white py-2 w-screen">
            DATA ABOUT POLLUTING GASES
          </p>
        </div>
      </div>
    </div>
  );
};

export default HomePage;
