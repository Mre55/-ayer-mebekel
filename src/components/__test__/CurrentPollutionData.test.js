import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import {
  fetchWeatherSuccess,
} from '../../redux/weather/weatherReducer';

import HomePage from '../HomePage';
import TopBar from '../TopBar';

describe('Dispatch actions tests', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);

  it('should dispatch fetch weather action', () => {
    const initialState = {};
    const store = mockStore(initialState);

    // Dispatch the action
    store.dispatch(fetchWeatherSuccess());

    // Test if your store dispatched the expected actions
    const actions = store.getActions();
    const expectedPayload = { type: 'FETCH_WEATHER' };
    expect(actions).toEqual([expectedPayload]);
  });
});

describe('Home page component snapshot tests', () => {
  const mockStore = configureStore();

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <HomePage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <TopBar />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});