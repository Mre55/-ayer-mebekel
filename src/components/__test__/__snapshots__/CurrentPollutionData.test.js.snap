// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component snapshot tests renders correctly 1`] = `
<div>
  <div
    className=""
  >
    <div
      className="flex bg-[#5788e6] w-screen text-white px-3 items-center md: justify-evenly"
    >
      <h2
        className="text-3xl font-bold"
      >
        Forecast air pollution data
      </h2>
      <div
        className="text-center py-4"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="latitude"
          >
            <p
              className="text-lg"
            >
              Latitude
            </p>
            <input
              className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
              id="latitude"
              name="lat"
              onChange={[Function]}
              placeholder="Enter a Latitude..."
              required={true}
              type="number"
            />
            <br />
          </label>
          <label
            htmlFor="longitude"
          >
            <p
              className="text-lg"
            >
              Longitude
            </p>
            <input
              className="border-solid border-2 border-gray-50 my-1 rounded-full text-center text-black"
              id="longitude"
              name="long"
              onChange={[Function]}
              placeholder="Enter a Longitude..."
              required={true}
              type="number"
            />
            <br />
          </label>
          <input
            className="bg-blue-600 text-white w-20 text-lg rounded mt-4"
            onClick={[Function]}
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="w-screen"
      >
        <h2
          className="bg-[#35538c] text-sm text-white md:text-center pl-4 py-1"
        >
          CURRENT AIR POLLUTION DATA
        </h2>
        <div
          className="grid grid-cols-2"
        >
          <div
            className="flex flex-col items-end bg-[#4269b0] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                ADDIS ABABA
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
          <div
            className="flex flex-col items-end bg-[#3d60a3] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                HAWASSA
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
        </div>
        <div
          className="grid grid-cols-2"
        >
          <div
            className="flex flex-col items-end bg-[#3d60a3] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className="text-lg font-bold font-bold"
              >
                HOSSANA
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
          <div
            className="flex flex-col items-end bg-[#4269b0] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                LONDON
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
        </div>
        <div
          className="grid grid-cols-2"
        >
          <div
            className="flex flex-col items-end bg-[#4269b0] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                ADDIS ABABA
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
          <div
            className="flex flex-col items-end bg-[#3d60a3] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                HAWASSA
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
        </div>
        <div
          className="grid grid-cols-2"
        >
          <div
            className="flex flex-col items-end bg-[#3d60a3] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                HOSSANA
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
          <div
            className="flex flex-col items-end bg-[#4269b0] h-44 gap-8 p-4"
          >
            <button
              className="w-7"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="detail"
                src="detailIcon.png"
              />
            </button>
            <div
              className="text-white flex flex-col items-end"
            >
              <h4
                className=" text-lg font-bold font-bold"
              >
                LONDON
              </h4>
              <p>
                 
                Air Quality:
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hidden"
  >
    <div>
      <div
        className="flex items-center gap-14 bg-[#35538c] text-white px-3 "
      >
        <button
          className="w-10"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="back icon"
            src="backIcon.png"
          />
        </button>
        <p
          className="text-sm text-white py-2 w-screen"
        >
          DATA ABOUT POLLUTING GASES
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home page component snapshot tests renders correctly 2`] = `
<nav
  className=""
>
  <div
    className="flex items-end justify-center py-2 ring-4 bg-[#4369b2]"
  >
    <h2
      className="text-2xl font-bold text-white"
    >
      አየር ብክለት 
    </h2>
    <h2
      className="text-lg font-bold text-white"
    >
       (Air pollution)
    </h2>
  </div>
</nav>
`;
